src/resources/js/store/localisation.js

```diff
@@ -6,11 +6,32 @@ export const state = {
 
 export const getters = {
     isInitialised: state => Object.keys(state.i18n).length > 0,
-    __: (state, getters, rootState, rootGetters) => (key) => {
-        const lang = rootGetters['preferences/lang'];
-        return state.i18n[lang]
+    __: (state, getters, rootState, rootGetters) => (key, params) => {
+        const lang = rootGetters['preferences/lang']
+            || ((rootState.preferences && rootState.preferences.global)
+                ? rootState.preferences.global.lang : null);
+
+        let translation = state.i18n[lang]
             ? state.i18n[lang][key]
             : key;
+
+        if (typeof translation === 'undefined'
+            && rootState.localisation.keyCollector) {
+                store.dispatch('localisation/addMissingKey', key);
+        }
+
+        translation = translation || key;
+        if(params) {
+            translation = translation.replace(/:(\w*)/g, function(e, key) {
+                let param = params[key.toLowerCase()];
+                if(key === key.toUpperCase()) // param is uppercased
+                    param = param.toUpperCase();
+                else if(key[0] === key[0].toUpperCase()) // first letter is uppercased
+                    param = param.charAt(0).toUpperCase() + param.slice(1);
+                return param;
+            });
+        }
+        return translation;
```

Comment:
Let's move all the `params` logic to the global helper

Reply:
The _global_ helper is not global, it can't be called from within the store (ex. for a Dialog); hence why it's placed here instead (actually globally reachable)

The helper (imho) should be just that, an easier access to the store's `__`; not containing extra logic unless it only made sense in that context

Reply:
see above

Reply:
For architectural reasons I definitely understand you don't want to call mutations/actions from getters (although this use-case is one of the few where it makes sense) but I don't see why you would move the params logic to the helper instead; otherwise you can't use parameterized translations from within the store (VueX is not a Vue component, see [Mixin docs](https://vuejs.org/v2/guide/mixins.html))

Reply:
I posted in the related issue the solution that will keep the store cleaner and solve your problem.

Reply:
>VueX is not a Vue component

sharing revelations I guess? :)

Reply:
> sharing revelations I guess? :)

No, clarifying why what you suggested wasn't the solution ;)

Reply:
This has implications for the page title btw:
- param names would just be shown instead of actual data
- (confirmed on laravel-enso.com) page titles show up as null or undefined when there's no translation

The _fix_ with these changes (importing the _external_ `__` from within localisation.js) feels like an anti-pattern, open to suggestions though